File: app/src/main/java/com/example/vladimir/easyenglishlearn/Answer.java
package com.example.vladimir.easyenglishlearn;

/**
 * Created by BOBAH on 26.03.2015.
 */
public class Answer {
    Word question;
    String answer;
    boolean translationDirection;

    public Answer(Word question, StringBuilder answerBuilder, boolean translationDirection) {
        super();
        this.question = question;
        this.answer = answerBuilder.toString();
        this.translationDirection =translationDirection;
    }

    boolean isAnswerCorrect(){
        return  translationDirection ? question.translation.equalsIgnoreCase(answer): question.lexeme.equalsIgnoreCase(answer);
    }

}

File: app/src/main/java/com/volodymyr_x/easyenglishlearn/model/Answer.kt
package com.volodymyr_x.easyenglishlearn.model

import com.volodymyr_x.easyenglishlearn.ui.model.WordUI

class Answer(
    private val question: WordUI?,
    private val answer: String,
    private val isLexemeToTranslation: Boolean
) {
    val isCorrect: Boolean
        get() = if (isLexemeToTranslation) {
            question?.translation.equals(answer, ignoreCase = true)
        } else {
            question?.lexeme.equals(answer, ignoreCase = true)
        }
}
